python/src/aica_api/client.py

```diff
@@ -40,30 +41,37 @@ def _endpoint(self, endpoint=''):
         """
         return f'{self._address}/v2/{endpoint}'
 
-    def _ws_endpoint(self, endpoint=''):
+    @staticmethod
+    def client_version() -> str:
         """
-        Build the connection address for a given websocket endpoint.
+        Get the version of this API client utility
 
-        :param endpoint: The websocket endpoint
-        :return: The constructed connection address
+        :return: The version of the API client
         """
-        return f'{self._ws_address}/{endpoint}'
+        return importlib.metadata.version('aica_api')
 
     def check(self) -> bool:
         """
-        Check if the API version is v2 (any v2.x.x tag)
+        Check if this API client is compatible with the detected API server version
+
+        :return: True if the client is compatible with the API server version, False otherwise
         """
-        # TODO: come up with a compatibility table in the future
         try:
             api_version = requests.get(f'{self._address}/version').json()
         except requests.exceptions.RequestException as e:
             print(f'Error connecting to the API! {e}')
             return False
-        new_version = api_version.startswith('2')
-        if not new_version:
-            print(f'The detected API version v{api_version} is older than the minimum API version v2.0.0 supported by '
-                  f'this client')
-        return new_version
+
+        if api_version.startswith('3'):
+            return True
+        elif api_version.startswith('2'):
+            print(f'The detected API version v{api_version} is older than the minimum API version v3.0.0 supported by '
+                  f'this client (v{self.client_version()}). Please install Python API client version v1.2.0 or later'
```

Comment:
Is `later` the correct word here? I would understand it as > 1.2.0

Reply:
Good point, since we know we aren't release anything between 1.2 and 2.0 we can make this more specific

Reply:
```suggestion
                  f'this client (v{self.client_version()}). Please install Python API client version v1.2.0 '
```